### PR TITLE
mgr/cephadm: remove keyring during daemon post-remove

### PR DESCRIPTION
**Pull request by mgfritch**
Remove keyrings during a daemon post-remove action.

After:
- [x] #36935

Fixes: https://tracker.ceph.com/issues/47298
Signed-off-by: Michael Fritch <mfritch@suse.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
needs rebase

**Comment by mgfritch** [comment]
@sebastian-philipp  rebased!

File: src/pybind/mgr/cephadm/services/cephadmservice.py [hunk 1]
```diff
@@ -216,7 +216,13 @@ def pre_remove(self, daemon_id: str) -> None:
         """
         Called before the daemon is removed.
         """
-        pass
+        logger.debug(f'Pre remove daemon {self.TYPE}.{daemon_id}')
+
+    def post_remove(self, daemon_id: str) -> None:
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/services/cephadmservice.py]: somehow I knew back then that we'll end up with a post_remove

File: src/pybind/mgr/cephadm/services/cephadmservice.py [hunk 2]
```diff
@@ -286,6 +290,18 @@ def get_config_and_keyring(self,
...
 
+    def remove_keyring(self, daemon: DaemonDescription):
+        daemon_id: str = daemon.daemon_id
+        host: str = daemon.hostname
+
+        entity = self.get_auth_entity(daemon_id, host=host)
+
+        logger.info(f'Remove keyring: {entity}')
+        ret, out, err = self.mgr.check_mon_command({
+            'prefix': 'auth rm',
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/services/cephadmservice.py:305]: what happens, if the keyring is missing. is `auth rm` going to fail or silently ignore?

**Review by sebastian-philipp — APPROVED** [review]

File: src/pybind/mgr/cephadm/services/cephadmservice.py (hunk 2 above)
Comment (mgfritch) [reply src/pybind/mgr/cephadm/services/cephadmservice.py:305]: correct, `auth rm` appears to be idempotent

**Comment by sebastian-philipp** [comment]
green: https://pulpito.ceph.com/swagner-2020-09-11_17:54:34-rados:cephadm-wip-swagner-testing-2020-09-11-1625-distro-basic-smithi/

**Pull request merged**